[16][MIG ] module_analysis

florian-dacosta opened the pull request:
Replace https://github.com/OCA/server-tools/pull/2608

Sorry I guess I did a wrong manipulation, I could not reopen the other after a force push.

@legalsylvain

legalsylvain committed:
[ADD] new module module_analysis

fixup! [ADD] new module module_analysis

fixup! fixup! [ADD] new module module_analysis

fixup! fixup! fixup! [ADD] new module module_analysis

fixup! fixup! fixup! fixup! [ADD] new module module_analysis

IMP exception message

fixup! fixup! fixup! fixup! fixup! [ADD] new module module_analysis

[REF] remove use of cloc. use pygount librairy instead

fixup! [REF] remove use of cloc. use pygount librairy instead

fixup! fixup! [REF] remove use of cloc. use pygount librairy instead

Apply suggestions from code review

Co-Authored-By: David Beal <david.beal@akretion.com>

Update module_analysis/views/menu.xml

Co-Authored-By: David Beal <david.beal@akretion.com>

Update module_analysis/tests/test_module.py

Co-Authored-By: David Beal <david.beal@akretion.com>

Update module_analysis/readme/CONFIGURE.rst

Co-Authored-By: David Beal <david.beal@akretion.com>

[IMP] handle encoding

[UPD] Update module_analysis.pot

[UPD] README.rst

[UPD] README.rst

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (35 of 35 strings)

Translation: server-tools-12.0/server-tools-12.0-module_analysis
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-module_analysis/zh_CN/

[UPD] README.rst

santostelmo committed:
[IMP] module_analysis: black, isort, prettier

santostelmo committed:
[MIG] module_analysis: Migration to 13.0

santostelmo committed:
Fix pre-commit warnings

santostelmo committed:
fixup! Fix pre-commit warnings

santostelmo committed:
Skip _test modules from test_installed_modules

santostelmo committed:
Remove anonymous variable _

santostelmo committed:
fixup! [MIG] module_analysis: Migration to 13.0

santostelmo committed:
[Fix] author_txt_list assign

santostelmo committed:
Fix _get_files_to_analyse

legalsylvain committed:
[FIX+IMP] module_analysis : remove analysis when updating modules, because the analysis is partial (it also make the update slower) ; Add instead a cron task that is executed nightly to update analysis automatically

alessandro-fiorino committed:
Added translation using Weblate (Italian)

alessandro-fiorino committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (35 of 35 strings)

Translation: server-tools-13.0/server-tools-13.0-module_analysis
Translate-URL: https://translation.odoo-community.org/projects/server-tools-13-0/server-tools-13-0-module_analysis/it/

kevinkhao committed:
[IMP] Add SCSS SLOC measure

kevinkhao committed:
[FIX] fix usage of cron data, deactivate it by default

legalsylvain committed:
[REF] module_analysis: add legalsylvain as maintainers

royle-vietnam committed:
[FIX] module_analysis: correct syntax

Commit:
[FIX] test: proper analysis before testing

Commit:
[MIG] module_analysis: Migration to 15.0

[MIG] module_analysis: Migration to 15.0

Commit:
[UPD] Update module_analysis.pot

OCA-git-bot committed:
[UPD] README.rst

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 97.2% (35 of 36 strings)

Translation: server-tools-15.0/server-tools-15.0-module_analysis
Translate-URL: https://translation.odoo-community.org/projects/server-tools-15-0/server-tools-15-0-module_analysis/it/

florian-dacosta committed:
[IMP] module_analysis: black, isort, prettier

florian-dacosta committed:
[16.0][MIG] module_analysis : migrate to 16

florian-dacosta committed:
[IMP] Remove automatic analyse at installation and update
Analyse can take some time and a cron task can do it every day if needed, no need to recompute it in real time

florian-dacosta committed:
[FIX] Playing test after install must not be mandatory anymore because there is no post install hook anymore

florian-dacosta commented:
@legalsylvain 
>>Why [8e5457e](https://github.com/OCA/server-tools/commit/8e5457e805d46c7e65fd9eb2a57189cb504660a6)

>In general I prefer all cron to be unactive at installation, mainly this kind of cron that you have to do some configuration (better change the time to run it at night at least, etc). It does not change a lot to unactivate it because you have to configure it anyway.
Having active cron of the kind is annoying for test database, CI, preprod, etc...

You'll tell me if it is acceptable for you, else I'll remove the commit so we can go forward

legalsylvain commented:
I have no clear point of view. 
In one hand, you removed the analysis in the post_install part, and you set active to false for the cron. So when we install the module, now, nothing "work", (and will work) and the configuration is now mandatory.
In the other hand, your argument are valid. ;-)

So, it's as you wish on that topic ! 
But if we go in that direction, you should rewrite some readme files, because some part are obsolete / wrong now.

florian-dacosta committed:
module_analysis : Update readme

florian-dacosta commented:
Thanks @legalsylvain I always forget to check the readme files...
I've updated it the readme file and I also added the possibility to add a custom domain in the cron, because we have a use case where we'd like the analysis to run on uninstalled modules. (I will allow to know the "cost" of your module before installing it...)

legalsylvain commented on module_analysis/readme/CONFIGURE.rst:
```diff
@@ -16,9 +16,9 @@ developped for exemple,
-* Check the box 'Analyse Installed modules'
+* Run manually the action : 'Update Module Analysis'
 
     .. image:: ../static/description/base_module_update.png
```
I guess that line has to be removed. (and the according png file).

florian-dacosta replied:
I've added the line => To run an update manually, you can run the scheduled action manually.
The image is indeed to remove

legalsylvain commented:
> (I will allow to know the "cost" of your module before installing it...)

good idea !

florian-dacosta committed:
[IMP] module_analysis : allow to pass a custom domain to the analysis cron.
Usefull if we want to analyse the code for uninstalled modules for instance

PierrickBrun reviewed (approved):
looks good, I tested it in a customer project and the analysis worked well

florian-dacosta commented:
@legalsylvain 
Would you mind updating your review ?

legalsylvain reviewed (approved):
Thanks ! 

/ocabot merge nobump

bguillot commented:
/ocabot merge nobump

bguillot commented:
Thanks !
